File: app/src/main/java/org/michaelbel/tjgram/data/entity/Subsite.java
package org.michaelbel.tjgram.data.entity;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Subsite implements Serializable {
    public static final int TYPE_COMPANY = 3;
    public static final int TYPE_SECTION = 2;
    public static final int TYPE_USER = 1;

    @Expose @SerializedName("avatar_url") public String avatarUrl;
    @Expose @SerializedName("comments_count") public int commentsCount;
    @Expose @SerializedName("contacts") public SubsiteContacts contacts;
    //@Expose @SerializedName("cover") public SubsiteCover cover;
    @Expose @SerializedName("created") public long created;
    @Expose @SerializedName("createdRFC") public String createdRFC;
    @Expose @SerializedName("description") public String description;
    @Expose @SerializedName("entries_count") public int entriesCount;
    @Expose @SerializedName("id") public long id;
    @Expose @SerializedName("is_enable_writing") public boolean isEnableWriting;
    @Expose @SerializedName("is_muted") public boolean isMuted;
    @Expose @SerializedName("is_subscribed") public boolean isSubscribed;
    @Expose @SerializedName("is_unsubscribable") public boolean isUnsubscribable;
    @Expose @SerializedName("is_verified") public boolean isVerified;
    @Expose @SerializedName("karma") public int karma;
    @Expose @SerializedName("name") public String name;
    @Expose @SerializedName("rules") public String rules;
    @Expose @SerializedName("subscribers_count") public int subscribersCount;
    @Expose @SerializedName("type") public long type;
    @Expose @SerializedName("url") public String url;
    @Expose @SerializedName("vacancies_count") public int vacanciesCount;
}

File: app/src/main/java/org/michaelbel/tjgram/data/entity/Subsite.kt
package org.michaelbel.tjgram.data.entity

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

data class Subsite (
    @Expose @SerializedName("avatar_url") var avatarUrl: String,
    @Expose @SerializedName("comments_count") var commentsCount: Int,
    @Expose @SerializedName("contacts") var contacts: SubsiteContacts,
    //@Expose @SerializedName("cover") var cover: SubsiteCover,
    @Expose @SerializedName("createdDate") var created: Long,
    @Expose @SerializedName("createdDateRFC") var createdRFC: String,
    @Expose @SerializedName("description") var description: String,
    @Expose @SerializedName("entries_count") var entriesCount: Int,
    @Expose @SerializedName("id") var id: Long,
    @Expose @SerializedName("is_enable_writing") var isEnableWriting: Boolean,
    @Expose @SerializedName("is_muted") var isMuted: Boolean,
    @Expose @SerializedName("is_subscribed") var isSubscribed: Boolean,
    @Expose @SerializedName("is_unsubscribable") var isUnsubscribable: Boolean,
    @Expose @SerializedName("is_verified") var isVerified: Boolean,
    @Expose @SerializedName("karma") var karma: Int,
    @Expose @SerializedName("name") var name: String,
    @Expose @SerializedName("rules") var rules: String,
    @Expose @SerializedName("subscribers_count") var subscribersCount: Int,
    @Expose @SerializedName("type") var type: Long,
    @Expose @SerializedName("url") var url: String,
    @Expose @SerializedName("vacancies_count") var vacanciesCount: Int
): Serializable